source/data-handling-database-R.Rmd

```diff
@@ -0,0 +1,150 @@
+
+You should see a list similar to the list underneath, with the names of the available DSN names enlisted:
+![odbc-connecties](../img/odbc_gegevensbron.png)
+
+An alternative way to check the DSN name of a database already working on with Access, is to check the DSN inside MS Access (in dutch, check menu item *Koppelingsbeheer*): 
+![access-dsn](../img/access_dsn.png)
+
+For example, the DSN name `UserFlora` or `Cydonia-prd` can be used to query these databases and extract data from it with similar queries to the one used in MSAccess. First of all, the connection with the database need to be established, by using the `odbcConnect` function, providing the DSN name as argument:
+
+```{r sqlconnection}
+# WINDOWS USERS (uncomment when working in Windows)
+#my_connection <- odbcConnect("UserFlora") # uncomment for windows users
+
+# LINUX/MAC USERS (comment out when working in Windows)
+my_connection <- odbcDriverConnect("Driver={ODBC Driver 13 for SQL Server};Server=inbosql04.inbo.be;Database=UserFlora;Trusted_Connection=yes;")
+```
+
+Once this connection is sucessfully established, the database can be queried. 
+
+## Get a complete table from the databse
+
+The function `sqlFetch` can be used to load an entire table from a database. For example, to extract the `tblTaxon` table from the flora database:
+
+```{r querytable}
+rel_taxa <- sqlFetch(my_connection, "dbo.relTaxonTaxonGroep")
+head(rel_taxa)
+```
+
+with the connection `my_connection` made earlier is used as the first argument, the table name is the second argument.
+
+**Remark:** If you have no idea about the size of the table you're trying to load from the database, this could be rather tricky and cumbersome. Hence, it is probably better to only extract a portion of the table using a query.
+
+## Execute a query to the database
+
+The function `sqlQuery` provides more flexibilty as it can be used to try any SQL-query on the database. A complete introduction to the SQL language is out of scope here. We will focus on the application and the reusage of a query.
+
+```{r queryrun}
+meting <- sqlQuery(my_connection, paste("SELECT TOP 10 * FROM dbo.tblMeting", 
+                                        "WHERE Cor_X IS NOT NULL"))
+head(meting)
+```
+
+## Create and use query templates
+
+When you regularly use similar queries, with some minimal alterations, you do not want to copy/paste each time the entire query. It is prone to errors and you're script will become verbose. It is advisable to create query *templates*, that can be used within the `sqlQuery` function. 
+
+Consider the execution of the following query. We are interested in those records with valid X and Y coordinates for the measurement, based on a given dutch name:
+
+```{r queryexample}
+subset_meting <- sqlQuery(my_connection, 
+"SELECT meet.Cor_X
+	 , meet.COr_Y
+	 , meet.MetingStatusCode
+	 , tax.NaamNederlands
+	 , tax.NaamWetenschappelijk
+	 , waar.IFBLHokID
+FROM  [dbo].[tblMeting] meet
```

Comment:
[dbo].[tblMeting] **AS** meet

I prefer to be a bit verbose as that is more readable, especially for novices.